checker/src/org/checkerframework/checker/linear/LinearVisitor.java

```diff
@@ -69,6 +75,21 @@ public Void visitMemberSelect(MemberSelectTree node, Void p) {
         return super.visitMemberSelect(node, p);
     }
 
+    @Override
+    protected void commonAssignmentCheck(
+            AnnotatedTypeMirror varType,
+            AnnotatedTypeMirror valueType,
+            Tree valueTree,
+            String errorKey) {
+        if (varType.hasAnnotation(Linear.class)) {
+            if (valueTree instanceof LiteralTree || valueTree instanceof NewClassTree) {
+                valueType.removeAnnotation(atypeFactory.NORMAL);
+                valueType.addAnnotation(atypeFactory.LINEAR);
```

Comment:
If valueType has some other annotation beside `@Normal`, then it will have two annotations.  (An annotated type can only have one annotation per hierarchy.) Use replaceAnnotation instead.